lib/file_transfer_agent/encrypt_util.js

```diff
@@ -125,206 +124,278 @@ function EncryptUtil(encrypt, filestream, temp) {
     return newMatDesc;
   }
 
+  function createEncryptionMetadata(encryptionMaterial, keySize, encryptedKey, dataIv, keyIv = null, dataAad = null, keyAad = null) {
+    const matDesc = MaterialDescriptor(
+      encryptionMaterial.smkId,
+      encryptionMaterial.queryId,
+      keySize * 8
+    );
+
+    if (keyIv && dataAad && keyAad) {
+      keyIv = keyIv.toString(BASE64);
+      dataAad = dataAad.toString(BASE64);
+      keyAad = keyAad.toString(BASE64);
+    }
+
+    return new EncryptionMetadata(
+      encryptedKey.toString(BASE64),
+      dataIv.toString(BASE64),
+      matDescToUnicode(matDesc),
+      keyIv,
+      dataAad,
+      keyAad
+    );
+  }
+
   /**
-  * Encrypt file stream using AES algorithm.
-  *
-  * @param {Object} encryptionMaterial
-  * @param {String} fileStream
-  * @param {String} tmpDir
-  * @param {Number} chunkSize
-  *
-  * @returns {Object}
-  */
-  this.encryptFileStream = async function (encryptionMaterial, fileStream) {
-    // Get decoded key from base64 encoded value
-    const decodedKey = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
-    const keySize = decodedKey.length;
+   * Encrypt content using AES-CBC algorithm.
+   */
+  this.encryptFileStream = async function (encryptionMaterial, content) {
+    return this.encryptDataCBC(encryptionMaterial, content);
+  };
+
+  this.encryptDataCBC = function (encryptionMaterial, data) {
+    const decodedKek = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
+    const keySize = decodedKek.length;
 
-    // Get secure random bytes with block size
-    const ivData = getSecureRandom(blockSize);
+    const dataIv = getSecureRandom(AES_CBC.ivSize);
     const fileKey = getSecureRandom(keySize);
 
-    // Create cipher with file key, AES CBC, and iv data
-    let cipher = crypto.createCipheriv(aesCbc(keySize), fileKey, ivData);
-    const encrypted = cipher.update(fileStream);
-    const final = cipher.final();
-    const encryptedData = Buffer.concat([encrypted, final]);
+    const dataCipher = crypto.createCipheriv(AES_CBC.cipherName(keySize), fileKey, dataIv);
+    const encryptedData = performCrypto(dataCipher, data);
 
-    // Create key cipher with decoded key and AES ECB
-    cipher = crypto.createCipheriv(aesEcb(keySize), decodedKey, null);
+    const keyCipher = crypto.createCipheriv(AES_ECB.cipherName(keySize), decodedKek, null);
+    const encryptedKey = performCrypto(keyCipher, fileKey);
 
-    // Encrypt with file key
-    const encKek = Buffer.concat([
-      cipher.update(fileKey),
-      cipher.final()
-    ]);
+    return {
+      encryptionMetadata: createEncryptionMetadata(encryptionMaterial, keySize, encryptedKey, dataIv),
+      dataStream: encryptedData
+    };
+  };
 
-    const matDesc = MaterialDescriptor(
-      encryptionMaterial.smkId,
-      encryptionMaterial.queryId,
-      keySize * 8
-    );
+  //TODO: Add proper usage when feature is ready (SNOW-940981)
```

Comment:
let's use one format of todos - above there is `TODO: SNOW-XXX: ....` also in other places

Reply:
Done